mne/time_frequency/tests/test_psd.py

```diff
@@ -14,99 +16,102 @@
+    evoked = epochs.average()
 
     tmin_full, tmax_full = -1, 1
-    epochs_full = Epochs(raw, events[:10], event_id, tmax=tmax_full,
-                         tmin=tmin_full, picks=picks,
-                         baseline=(None, 0),
+    epochs_full = Epochs(raw, events[:10], event_id, tmin_full, tmax_full,
+                         picks=picks, baseline=(None, 0),
                          reject=dict(grad=4000e-13, eog=150e-6), proj=False,
                          preload=True)
 
-    picks = pick_types(epochs.info, meg='grad', eeg=False, eog=True,
-                       stim=False, include=include, exclude='bads')
-    psds, freqs = compute_epochs_psd(epochs[:1], fmin=2, fmax=300,
-                                     n_fft=n_fft, picks=picks)
-
-    psds_t, freqs_t = compute_epochs_psd(epochs_full[:1], fmin=2, fmax=300,
-                                         tmin=tmin, tmax=tmax,
-                                         n_fft=n_fft, picks=picks)
-    # this one will fail if you add for example 0.1 to tmin
-    assert_array_almost_equal(psds, psds_t, 27)
-
-    psds_proj, _ = compute_epochs_psd(epochs[:1].apply_proj(), fmin=2,
-                                      fmax=300, n_fft=n_fft, picks=picks)
-
-    assert_array_almost_equal(psds, psds_proj)
-    assert_true(psds.shape == (1, len(picks), len(freqs)))
-    assert_true(np.sum(freqs < 0) == 0)
-    assert_true(np.sum(psds < 0) == 0)
+    kws_psd = dict(tmin=tmin, tmax=tmax, fmin=fmin, fmax=fmax,
+                   picks=picks_psd)  # Common to all
+    funcs = [(psd_welch, kws_welch),
+             (psd_multitaper, kws_mt),
+             (compute_epochs_psd, kws_welch)]
+
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter('always')
+        for func, kws in funcs:
+            kws = kws.copy()
+            kws.update(kws_psd)
+
+            psds, freqs = func(
+                epochs[:1], proj=False, **kws)
+            psds_proj, freqs_proj = func(
+                epochs[:1], proj=True, **kws)
+            psds_f, freqs_f = func(
+                epochs_full[:1], proj=False, **kws)
+
+            # this one will fail if you add for example 0.1 to tmin
+            assert_array_almost_equal(psds, psds_f, 27)
+            assert_array_almost_equal(psds, psds_proj, 27)
+
+            assert_true(psds.shape == (1, len(kws['picks']), len(freqs)))
+            assert_true(np.sum(freqs < 0) == 0)
+            assert_true(np.sum(psds < 0) == 0)
+
+            # Array input shouldn't work
+            assert_raises(ValueError, func, epochs.get_data())
+
+            if func is not compute_epochs_psd:
+                # Testing evoked (doesn't work w/ compute_epochs_psd)
+                psds_ev, freqs_ev = func(
+                    evoked, proj=False, **kws)
+                psds_ev_proj, freqs_ev_proj = func(
+                    evoked, proj=True, **kws)
+                assert_array_almost_equal(psds_ev, psds_ev_proj, 27)
```

Comment:
This is a test that I copied over from the previous version of the code. I didn't apply proj anywhere else, but it will be applied within the function because of the `proj=True` flag in the second case. I'm not an MEG guy so I have no idea whether or not this is a reasonable check...I just tried to change as little as possible :)